src/stores/store.c

```diff
@@ -61,27 +61,30 @@ LabelStore *LabelStore_New(RedisModuleCtx *ctx, LabelStoreType type, const char
 
 LabelStore *LabelStore_Get(RedisModuleCtx *ctx, LabelStoreType type, const char *graph, const char* label) {
 	LabelStore *store = NULL;
-    char *strKey;
-    LabelStore_Id(&strKey, type, graph, label);
-
-    RedisModuleString *rmStoreId = RedisModule_CreateString(ctx, strKey, strlen(strKey));
-    free(strKey);
-    
-	RedisModuleKey *key = RedisModule_OpenKey(ctx, rmStoreId, REDISMODULE_WRITE);
-    RedisModule_FreeString(ctx, rmStoreId);
-
-    /* Create "ALL" store if doesn't exists. */
-    if((RedisModule_KeyType(key) == REDISMODULE_KEYTYPE_EMPTY) && label == NULL) {
-        return LabelStore_New(ctx, type, graph, "ALL", -1);
-    }
-
-    if (RedisModule_ModuleTypeGetType(key) != StoreRedisModuleType) {
-        return NULL;
-	}
-
-	store = RedisModule_ModuleTypeGetValue(key);
-    RedisModule_CloseKey(key);
-	return store;
+  char *strKey;
+  LabelStore_Id(&strKey, type, graph, label);
+
+  RedisModuleString *rmStoreId = RedisModule_CreateString(ctx, strKey, strlen(strKey));
+  free(strKey);
+
+  RedisModuleKey *key = RedisModule_OpenKey(ctx, rmStoreId, REDISMODULE_WRITE);
+  RedisModule_FreeString(ctx, rmStoreId);
+
+  if((RedisModule_KeyType(key) == REDISMODULE_KEYTYPE_EMPTY)) {
+      RedisModule_CloseKey(key);
```

Comment:
The only big change in this function is to properly close the key - it squashes a Helgrind warning, though probably not a race condition.